cli/cmd/generate.go

```diff
@@ -415,21 +393,11 @@ func addSeedshareOwnerKeyToManifest(manifst *manifest.Manifest, keyPath string)
 	if err != nil {
 		return fmt.Errorf("reading seedshare owner key: %w", err)
 	}
-	block, _ := pem.Decode(keyData)
-	if block == nil {
-		return errors.New("failed to decode PEM block")
-	}
-	var publicKey manifest.HexString
-	switch block.Type {
-	case "RSA PRIVATE KEY":
-		privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
-		if err != nil {
-			return fmt.Errorf("parsing RSA private key: %w", err)
-		}
-		publicKey = manifest.MarshalSeedShareOwnerKey(&privateKey.PublicKey)
-	default:
-		return fmt.Errorf("unsupported PEM block type: %s", block.Type)
+	privateKey, err := manifest.ParseSeedshareOwnerPrivateKey(keyData)
```

Comment:
Could this be more similar to the workload owner key function, like `ExtractSeedshareOwnerPublicKey`?